dash/dash.py

```diff
@@ -43,19 +87,34 @@ def __init__(
                 ''', DeprecationWarning)
 
         name = name or 'dash'
+
+        self.assets_folder = assets_folder or os.path.join(
+            os.getcwd(), 'assets'
```

Comment:
I'm a little worried that the use of `os.getcwd()` could be a little brittle. The `name` param to the Flask instance is mandatory so that Flask can derive the root for place the relative `static_folder`. If we use getcwd and Flask is always placing its static folder relative to the user's package/module then there is the possibility for the root of our assets location and the root of Flask's static folder locations to become out of synch (eg with `os.chdir()`). Does that make any sense?

If this is a real issue I'm wondering if it would be better to tap into however Flask derives the root location from the `name` param?

Reply:
I investigated what flask does, it get the root directory from `flask.helpers.get_root_path`. It first try to get it from `sys.modules`, if it's not found, it will do `os.getcwd` as a fallback. That is the reason it takes a required arg and generally you do `app = Flask(__name__)`, this will give the  directory of the file that contain the app instance in every case. 

But dash do `name='__main__'` by default, which can result in unexpected behavior from what I've tested. If the app was defined on the root folder, but it uses a script in a sub folder to start the server, it will have the wrong directory. Or if you run the app with the command `python -m flask run`, it will be `/venv/lib/site-packages/flask`. I think it also applies to wsgi runned instances, the `__main__` file won't be the app file but the wsgi runner. 

We could use `flask.helpers.get_root_path` instead of `os.getcwd`, but I think we should require the `__name__` of the file like flask does to ensure consistency, but it would break compatibility with older version as now we would have a required arg in the dash constructor.

Reply:
In a subsequent PR, maybe we can make `name` a keyword argument and then include usage in the docs?

Reply:
Ah that's interesting. Changing `name` to default to `__main__` was my change. I didn't test it as thoroughly as that, so missed those issues. So while it was an improvement on the previous default of `dash`, it's still got issues. Well spotted!

Not sure what you mean @chriddyp?  `name` currently is a keyword argument of `Dash`.

I don't think we can hope to intelligently infer the correct value of `name`, as the challenges around that are precisely why Flask makes this param mandatory, as @T4rk1n points out. So perhaps documentation is the best way around those edge cases.